clients/java/zts/core/src/main/java/com/yahoo/athenz/zts/ZTSFileUtil.java

```diff
@@ -0,0 +1,105 @@
+import com.yahoo.athenz.auth.token.AccessToken;
+import com.yahoo.athenz.auth.token.jwts.JwtsSigningKeyResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+public class ZTSFileUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZTSFileUtil.class);
+
+    static private final String DEFAULT_ACCESS_TOKEN_DIR_PATH = "/var/lib/sia/tokens/";
+    static private final String ACCESS_TOKEN_PATH_PROPERTY = "zts.client.accesstoken.path";
+    static private final String ROLE_NAME_CONNECTOR = ",";
+    private JwtsSigningKeyResolver accessSignKeyResolver = null;
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private Map<String, String> roleNameMap;
+
+    public ZTSFileUtil(JwtsSigningKeyResolver resolver) {
+        roleNameMap = new HashMap<>();
+        accessSignKeyResolver = resolver;
+    }
+
+    public void preload() {
+        String path = System.getProperty(ACCESS_TOKEN_PATH_PROPERTY, DEFAULT_ACCESS_TOKEN_DIR_PATH);
+        File dir = new File(path);
+
+        // preload the map from the domain, rolesname to file path
+        if (dir.exists() && dir.isDirectory()) {
+            for (File domainDir: dir.listFiles()) {
+                if (domainDir.isDirectory()) {
+                    for (File tokenFile: domainDir.listFiles()) {
+                        if (!tokenFile.isDirectory()) {
+                            AccessTokenResponse accessTokenResponse = null;
+                            try {
+                                accessTokenResponse = objectMapper.readValue(tokenFile, AccessTokenResponse.class);
+                            } catch (IOException e) {
+                                LOG.warn("Failed to load or parse token file: {}", tokenFile);
+                            }
+
+                            if (accessTokenResponse == null) {
+                                continue;
+                            }
+
+                            AccessTokenResponseCacheEntry cacheEntry = new AccessTokenResponseCacheEntry(accessTokenResponse);
+
+                            if (!cacheEntry.isExpired(-1)) {
+                                try {
+                                    addToRoleMap(domainDir.getName(), tokenFile.getName(), accessTokenResponse);
+                                } catch (IOException e) {
+                                    LOG.warn("Failed to put role map for token: {}", tokenFile.getName());
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    }
+
+    // function load the access token from file
+    public AccessTokenResponse lookupAccessTokenFromDisk(String domain, List<String> rolesName) throws IOException {
+        String path = System.getProperty(ACCESS_TOKEN_PATH_PROPERTY, DEFAULT_ACCESS_TOKEN_DIR_PATH);
```

Comment:
make this and other String variables as final as necessary.

Reply:
updated